### PR TITLE
Allow wildcards with faf repo manipulation

### PR DESCRIPTION
**Pull request by michalfabik**
Allow multiple `REPO` arguments to `faf repomod` ~~and `faf repoinfo`~~, using either enumeration or wildcards.  
~~Change `faf repoinfo` output format for easier parsing.~~  
Add OS and architecture info to `faf repolist` output.
Add a short `-a` option as an alternative to `--all` to `faf repodel`.

### CODE REVIEW

**Comment by mkutlak** [comment]
Could you add a commentary to the commit messages describing what it adds/changes (with examples)?

File: tests/test_actions.py [hunk 1]
```diff
@@ -234,6 +234,12 @@ def repomod_testing(self, repo_type):
         self.assertNotIn("file:///sample_rpms", self.action_stdout)
         self.assertIn("file:///some/new/url", self.action_stdout)
 
+        self.assertEqual(self.call_action_ordered_args("repomod", [
+            "sample_repo", "another_repo", "--gpgcheck enable"]), 0)
```
Comment (mkutlak) [inline tests/test_actions.py]: You should also check whether the changes are actually applied with `assertNotIn`/`assertIn`, or with whatever other assert there is.

**Comment by michalfabik** [comment]
> Could you add a commentary to the commit messages describing what it adds/changes (with examples)?  
  
Commit messages updated.

File: tests/test_actions.py (hunk 1 above)
Comment (michalfabik) [reply tests/test_actions.py]: Added.

**Comment by mkutlak** [comment]
@trapas retest this please

**Comment by mkutlak** [comment]
What about https://pypi.org/project/tabulate/ to create the repoinfo table?

Its available in Fedora and EPEL8.
https://koji.fedoraproject.org/koji/packageinfo?packageID=21436

**Comment by michalfabik** [comment]
> What about https://pypi.org/project/tabulate/ to create the repoinfo table?  
  
Yeah, I was actually looking into that one (and a couple of other similar libraries) but then I felt hesitant about introducing a new dependency. I thought the repolist output is unlikely to ever get a lot more complex than it is now, so doing it all by hand is probably still acceptable.  
I can rewrite it if you insist, of course.

**Comment by mkutlak** [comment]
I've looked where else we could use the `tabulate` and saw `repolist --detailed`.

``` sh
[root@e94ae2a3b269 actions]# faf repolist --detailed
Name      Type  URL          Nice name  
f31       dnf   file:///tmp             
                asdf                    
f32       dnf   file:///tmp             
f32_??    dnf   file:///tmp             
f32_**??  dnf   file:///tmp             
hello     dnf   file:///tmp             
                asdf                    

[root@e94ae2a3b269 actions]# faf repoinfo --all
NAME            NICE_NAME       TYPE    URL                     GPG_CHECK
f31                             dnf     file:///tmp, asdf       False    
f32                             dnf     file:///tmp             False    
f32_**??                        dnf     file:///tmp             True     
f32_??                          dnf     file:///tmp             True     
hello                           dnf     file:///tmp, asdf       True
```

IMO, just drop the `repoinfo` changes and improve `repolist` instead (assigned release/opsys, gpgcheck, ...).

File: src/pyfaf/actions/repomod.py [hunk 2]
```diff
@@ -29,13 +31,38 @@ def __init__(self):
...
+        if cmdline.all or '*' in cmdline.REPO:
+            repos.extend(db.session.query(Repo).all())
+
+        else:
+            for pattern in cmdline.REPO:
+                pattern = re.sub('_', r'\\_', pattern)
+                pattern = re.sub('%', r'\\%', pattern)
+                pattern = re.sub(r'\*', '%', pattern)
+                pattern = re.sub(r'\?', '_', pattern)
+                repos.extend(db.session.query(Repo).filter(Repo.name.like(pattern)).all())
```
Comment (mkutlak) [inline src/pyfaf/actions/repomod.py]: Can you make it a function (same code is also in `repodel`) ?

Comment (michalfabik) [reply src/pyfaf/actions/repomod.py]: Done.

**Comment by michalfabik** [comment]
> I've looked where else we could use the `tabulate` and saw `repolist --detailed`.
> 
> ```shell
> [root@e94ae2a3b269 actions]# faf repolist --detailed
> Name      Type  URL          Nice name  
> f31       dnf   file:///tmp             
>                 asdf                    
> f32       dnf   file:///tmp             
> f32_??    dnf   file:///tmp             
> f32_**??  dnf   file:///tmp             
> hello     dnf   file:///tmp             
>                 asdf                    
> 
> [root@e94ae2a3b269 actions]# faf repoinfo --all
> NAME            NICE_NAME       TYPE    URL                     GPG_CHECK
> f31                             dnf     file:///tmp, asdf       False    
> f32                             dnf     file:///tmp             False    
> f32_**??                        dnf     file:///tmp             True     
> f32_??                          dnf     file:///tmp             True     
> hello                           dnf     file:///tmp, asdf       True
> ```
> 
> IMO, just drop the `repoinfo` changes and improve `repolist` instead (assigned release/opsys, gpgcheck, ...).

True, rewritten.

**Comment by michalfabik** [comment]
@trapas retest this please

File: src/pyfaf/actions/repolist.py [hunk 3]
```diff
@@ -28,12 +28,26 @@ class RepoList(Action):
...
+                                           repo.opsys_list,
+                                           repo.opsysrelease_list,
+                                           repo.arch_list],
+                                          key=len))):
+                    data.append(("",
+                                 "",
+                                 repo.url_list[i].url if len(repo.url_list) > i else "",
+                                 "",
+                                 "",
+                                 repo.opsys_list[i] if len(repo.opsys_list) > i else "",
```
Comment (mkutlak) [inline src/pyfaf/actions/repolist.py]: I think you are not able to assign repo to OpSys only to OpSysRelease, so you can remove the `OS` column.

I've tried with 
```
$ faf repoassign hello Fedora
[2020-02-05 15:01:34] ERROR:faf.RepoAssign: Assigning operating system without release to non - parametrized repo. Assign operating system with release.
```

```
bash-5.0$ faf repolist --detailed
Name      Type  URL                                                                                                                                          Nice name  GPG check  OS  OS release  Architecture  
f31       dnf   file:///tmp                                                                                                                                             False          Fedora 32   x86_64        
                asdf                                                                                                                                                                                             
f32       dnf   file:///tmp                                                                                                                                             False                                    
                ftp://super-fricking-loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong-addresssssssssssssssssssssssssssssssssssss-ooooo                                                      
f32_??    dnf   file:///tmp                                                                                                                                             True                                     
f32_**??  dnf   file:///tmp                                                                                                                                             True                                     
hello     dnf   file:///tmp                                                                                                                                             True           Fedora 31   x86_64        
                asdf                                                                                                                                                                   Fedora 32   i386          
                                                                                                                                                                                       Fedora 30
```

**Comment by mkutlak** [comment]
I will merge this PR once the `OS` column is resolved.

File: src/pyfaf/actions/repolist.py (hunk 3 above)
Comment (michalfabik) [reply src/pyfaf/actions/repolist.py]: But the following works:  
  
    $ faf repoadd hello dnf file:///repo/releases/$  
    $ faf repoassign hello Fedora  
    $ faf repolist --detailed  
    Name          Type         URL                      Nice name  GPG check  OS      OS release  Architecture  
    hello         dnf          file:///repo/releases/$             True       Fedora  
  
Can't this ever happen in practice?

Comment (mkutlak) [reply src/pyfaf/actions/repolist.py]: Oh, you are right.

**Pull request merged**